components/vote/button/new-button.jsx

```diff
@@ -0,0 +1,75 @@
+import React, {Component} from 'react';
+import "./button-style.scss";
+
+export default class NewButton extends Component {
+  handleClick (e, n) {
+    const {maxUp, maxDown, onVote} = this.props;
+    onVote(Math.min(maxUp, Math.max(n, -maxDown)));
+    e.preventDefault();
+    return false;
+  }
+
+  titleText (n, maxUp, maxDown) {
+    n = Math.min(maxUp, Math.max(n, -maxDown));
+    if(n === 0)
+      return "";
+    return n > 0 ? "+" + n : "" + n;
+  }
+
+  makeTriangle (n, fn, size, minForEnabled) {
+    const {maxUp, maxDown, color} = this.props;
+    const enabled = n !== 0 && (n > 0 ? (maxUp >= minForEnabled) : (maxDown >= minForEnabled));
+    const className = "vote-new-button__upDown";
+    if(enabled) {
+      return <a href="#"
+        title={this.titleText(n, maxUp, maxDown)}
+        onClick={e => this.handleClick(e, n)}
+        className={className}>
+          {fn({size: size, color: color})}
+      </a>;
+    } else {
+      return <a
+        className={className}>
+          {fn({size: size, color: "#eee"})}
+      </a>;
+    }
+  }
+  render() {
+    const {voteAppToken} = localStorage;
```

Comment:
We should not let this component know about the login system (localStorage, voteAppToken). Could this information just be passed via props i. e loggedIn = true or myValue != null?